### DISCUSSION TITLE
Can not compile 1.5 on iOS

### DISCUSSION THREAD

**Discussion by JediThumper** [SchneiderInfosystems/FB4D#163] [Q&A]
Hi,

First thanks for all your efforts and updates.

I'm trying to get the new version of FB4D to compile on iOS.

I get an error: [DCC Error] FB4D.OAuth.pas(118): E2003 Undeclared identifier: 'SetKeyFromCert'
this occurs in function TTokenJWT.VerifySignature.

I gather its to do with RSA signing (the function is IFDEFed out)  in iOS but I can not find a solution.

Any help or suggestions would be appreciated.

Thanks

Alan

**Comment by SchneiderInfosystems** [comment]
Hi Alan

I'm traveling up to mid August and cannot check it.

I assume there was a change in the sub module JOSE JWT. Can you check if a change in this lib causes the problem. If so ask Pablo.

Thnx Christoph

**Comment by JediThumper** [comment]
Hi Christoph

Thanks for responding whilst on your travels. Have a great time.
The source has changed.
I have emailed Pablo. 
We'll wait to see what the response is.
Thanks again
Alan

**Comment by JediThumper** [comment]
Hi Christoph,
Hope you had a good holiday.
Paolo has added the issues to an existing thread: https://github.com/paolo-rossi/delphi-jose-jwt/issues/51

Can I ask does FB4D even need the function that is causing the trouble?

If not then we could safely comment it out?

Appreciate you input.

Alan

**Comment by SchneiderInfosystems** [comment]
Hi Alan

Yes, we realy enjoyed our longer holiday this year.

After reading the explanations from Paolo Rossi and Wagner Landgraf I decided to remove yet the TTokenJWT.VerifySignature functionality for iOS platform as long there is no solution. 

In fact, the functionality is still there also for iOS, but it always returns false because on iOS we cannot verify the JWT token retrieved from the Firebase Auth Service. This may reduce the security level of FB4D based applications but I don't see any other solution at the moment because we are at the end of the Indy-JOSE-FB4D dependency chain. Check the newest Comit of FB4D. I hope you can live with this in your project for now. 

Thanks for a short report if this makes everything compilable for iOS again, as I haven't tested this myself at the moment.

Christoph

**Comment by JediThumper** [comment]
Hi Christoph,

I have just downloaded a new copy of FB4D from github and added it to my project.

Unfortunately,  There was still an error in JOSE in... unit JOSE.OpenSSL.Headers;
From line 166
class function JoseSSL.LoadFunctionCLib(const AFunctionName: string; const ARaiseException: Boolean = True): Pointer;
begin
  **Result := {$IFDEF WINDOWS}Windows.{$ENDIF}GetProcAddress(GetCryptLibHandle, PChar(AFunctionName));**
  if Result = nil then
  begin
    if ARaiseException then
      raise Exception.CreateFmt('Error loading [%s]', [AFunctionName])
    else
      Inc(FLoadErrors);
  end;
end;

I have commented this line out to make everything compile. - Seems to work :)

Thanks

Alan
